src/main.rs

```diff
@@ -305,68 +300,78 @@ async fn handle_web_req(
                 .assume_no_inputs_seen_before(); // TODO
 
+            // `psbt` is going to ACTUALLY be the proposal PSBT
             let mut psbt = proposal.psbt().clone();
-            eprintln!("Received transaction: {:#?}", psbt);
-            {
-                for input in &mut psbt.unsigned_tx.input {
-                    // clear signature
-                    input.script_sig = bitcoin::blockdata::script::Script::new();
-                }
-            }
+            eprintln!("Received transaction: {:#?}", psbt); // but not yet
+            psbt.unsigned_tx
+                .input
+                .iter_mut()
+                .for_each(|txin| txin.script_sig = bitcoin::Script::new());
+
+            // find our output
+            // TODO: Would it be possible that the user sends to multiple outputs that we own?
             let (our_output, scheduled_payjoin) = handler
                 .payjoins
                 .find(&mut psbt.unsigned_tx.output)
                 .expect("the transaction doesn't contain our output");
-            let total_channel_amount: bitcoin::Amount = scheduled_payjoin
+
+            // TODO: make this a member fn of `ScheduledPayJoin`
+            // TODO: make sure it is impossible to create a BIP21pj linked to `ScheduledPayJoin`
+            //  with a "total channel amount" larger than requested + fee
+            let total_channel_amount = scheduled_payjoin
                 .channels
                 .iter()
                 .map(|channel| channel.amount)
                 .fold(bitcoin::Amount::ZERO, std::ops::Add::add);
+
+            // TODO: make this a member fn of `ScheduledPayJoin`
             let fees = calculate_fees(
                 scheduled_payjoin.channels.len() as u64,
                 scheduled_payjoin.fee_rate,
                 scheduled_payjoin.wallet_amount != bitcoin::Amount::ZERO,
             );
 
+            // TODO: not great
             assert_eq!(
                 our_output.value,
                 (total_channel_amount + scheduled_payjoin.wallet_amount + fees).as_sat()
             );
 
-            let chids = (0..scheduled_payjoin.channels.len())
+            let chan_ids = (0..scheduled_payjoin.channels.len())
                 .into_iter()
                 .map(|_| rand::random::<[u8; 32]>())
                 .collect::<Vec<_>>();
 
+            // these are channel-open txouts
             // no collect() because of async
             let mut txouts = Vec::with_capacity(scheduled_payjoin.channels.len());
 
-            for (channel, chid) in scheduled_payjoin.channels.iter().zip(&chids) {
+            // TODO: Creating `OpenChannelRequest` should be one thing, and async calls another
+            // TODO: Instead of interfacing with `tonic_lnd` directly, consider having a trait
+            //      so this can be extended in the future
+            // TODO: What if channel open fails or takes too long?
```

Comment:
We get a corrupt channel state 😬 This is the worst failure case in this code. Most important error to address.